app/src/main/java/com/example/bookwormadventuresdeluxe2/MyProfileFragment.java

```diff
@@ -71,24 +107,79 @@ public void onClick(View view)
                 case R.id.profile_edit:
                     final View editInfo = LayoutInflater.from(this.getContext()).inflate(R.layout.edit_profile, null);
 
-                    // Set up the input
-                    final EditText inputEmail = editInfo.findViewById(R.id.edit_email);
-                    final EditText inputPhone = editInfo.findViewById(R.id.edit_phone);
-                    // Specify the type of input expected
+                    /* Set up the input */
+                    EditText inputEmail = editInfo.findViewById(R.id.edit_email);
+                    EditText inputPhone = editInfo.findViewById(R.id.edit_phone);
+
+                    /* Specify the type of input expected */
                     inputEmail.setInputType(InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS);
                     inputPhone.setInputType(InputType.TYPE_CLASS_PHONE);
 
+                    /* Setting text to user's details */
+                    inputEmail.setText(viewUserObject.getEmail());
+                    inputPhone.setText(viewUserObject.getPhoneNumber());
+
+                    /* Create popup dialog for editing profile */
                     final AlertDialog builder = new AlertDialog.Builder(this.getContext()).create();
                     builder.setView(editInfo);
 
-                    // Set up the buttons
+                    /* Set up the buttons */
                     editInfo.findViewById(R.id.edit_confirm).setOnClickListener(new View.OnClickListener()
                     {
                         @Override
                         public void onClick(View view)
                         {
-                            //TODO: test input, get input, update user
-                            builder.dismiss();
+                            inputEmail.setError(null);
+                            inputPhone.setError(null);
+
+                            /* Checks if no changes were made */
+                            if (viewUserObject.getEmail().equals(inputEmail.getText().toString())
+                                && viewUserObject.getPhoneNumber().equals(inputPhone.getText().toString()))
+                            {
+                                builder.dismiss();
+                                return;
+                            }
+
+                            /* Checks if both email and phone number are empty and disables confirm button */
+                            if (TextUtils.isEmpty(inputEmail.getText().toString().trim())
```

Comment:
There is no reason for this duplicated code. Can you just make each of these if statements set some boolean instead of returning? Then at the bottom you can return if boolean true

Reply:
that is so both errors can show up at the same time and not just the first OR that breaks it

Reply:
its better for the user. If it was changed, it would break at email is empty and not show that phone is empty also

Reply:
/* Checks if error is present and disables confirm button */
                            if (inputEmail.getError() != null)
                            {
                                return;
                            }
                            else
                            {
                                /* Attempts to write new email and phone number */
                                FirebaseUserGetSet.changeAuthInfo(inputEmail,
                                                                    inputPhone,
                                                                    viewUserObject.getDocumentId());
                                if (inputEmail.getError() != null)
                                {
                                    /* Closes dialog */
                                    builder.dismiss();
                                }

if i moved one of these, it would fail after a failed attempt followed by another fail attempt

Reply:
both errors will still show up...

You need to do this:
instead of returning on each if statement you set a variable to True. At the end you return IF the variable is true. The flow will be the same but less duplicated code